### ISSUE TITLE
Running `mattermost version` requires writable config.json

### ISSUE THREAD

**Issue by octomike** [mattermost/mattermost-server#16756]
#### Summary
Running `mattermost version` as unprivileged user fails because config.json is not writable.

#### Steps to reproduce

```
sudo -u nagios /opt/mattermost/bin/mattermost version
```

#### Expected behavior

Output should print version information.

#### Observed behavior (that appears unintentional) 

After the latest upgrade to current ESR our icinga checks fail, because an unprivileged user is trying to figure out the mattermost version by running `mattermost version`, resulting in:

```
panic: failed to apply option: failed to apply Config option: failed to create store: unable to load on store creation: failed to persist: failed to write file: open /opt/mattermost/config/config.json: permission denied

goroutine 1 [running]:
github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.InitDBCommandContextCobra(0x427c2e0, 0x273d8bd, 0x11, 0x2713600)
	github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/init.go:18 +0x138
github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.versionCmdF(0x427c2e0, 0x42e26d8, 0x0, 0x0, 0x0, 0x0)
	github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/version.go:31 +0x83
github.com/spf13/cobra.(*Command).execute(0x427c2e0, 0x42e26d8, 0x0, 0x0, 0x427c2e0, 0x42e26d8)
	github.com/spf13/cobra@v1.0.0/command.go:842 +0x453
github.com/spf13/cobra.(*Command).ExecuteC(0x4277180, 0x0, 0x0, 0xc0000420b8)
	github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	github.com/spf13/cobra@v1.0.0/command.go:887
github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands.Run(...)
	github.com/mattermost/mattermost-server/v5/cmd/mattermost/commands/root.go:14
main.main()
	github.com/mattermost/mattermost-server/v5/cmd/mattermost/main.go:34 +0x87
```

**Comment by amyblais** [comment]
@octomike Would you be open to posting this on our troubleshooting forum here: https://forum.mattermost.org/? Our support team is more active there and can help determine what the cause of the issue is.

**Comment by octomike** [comment]
Hi @amyblais, thank you for reaching out.

Meanwhile we switched to the REST API to read the current server version and it's not an issue for us anymore. For what it's worth, I still consider this a bug or at least very surprising behaviour in the mattermost server utility.

**Comment by amyblais** [comment]
Thanks, I haven't been able to get feedback from devs on this yet, but I created a ticket for us to review https://mattermost.atlassian.net/browse/MM-32494.

**Comment by streamer45** [comment]
Hi @octomike,

could you provide the output of `ls -ls /opt/mattermost/config/config.json`?

**Comment by octomike** [comment]
Sure:

```
$ ls -ls /opt/mattermost/config/config.json
24 -rw-r--r-- 1 mattermost mattermost 20510 Jan 27 07:35 /opt/mattermost/config/config.json
```

**Comment by streamer45** [comment]
Thank you @octomike.

So there are two problems. First the `mattermost version` command should not write to `config.json`. This has been fixed recently (in https://github.com/mattermost/mattermost-server/pull/16805). Other than that you should probably execute CLI commands as the `mattermost` user since that's the owner of those files.

**Comment by octomike** [comment]
> This has been fixed recently (in #16805)

Cool, good to know. In that case, the issue really can be closed I think.

>  Other than that you should probably execute CLI commands as the `mattermost` user since that's the owner of those files.

I would concur for actions that actually modify or query restricted data. But as a sysadmin I feel this is really unexpected behavior, when all I want to do is get a help message or print some version data. But I understand that `mattermost version` might be doing more than just report a couple of numbers.

**Comment by amyblais** [comment]
Closing as the fix is now available in v5.31.1,
